tools/hardware/utils/memory.py

```diff
@@ -97,7 +97,7 @@ def get_physical_memory(tree: FdtParser, config: Config) -> List[Region]:
 def get_addrspace_exclude(regions: List[Region], config: Config):
     ''' Returns a list of regions that represents the inverse of the given region list. '''
     ret = set()
-    as_max = utils.align_down(config.addrspace_max, config.get_page_bits())
+    as_max = utils.align_down(config.addrspace_max, config.get_smallest_kernel_object_alignment())
```

Comment:
This should get a comment that we deliberately want to exclude the very last part of the address space here.

Reply:
We aren't excluding the last part of the address space, but making sure that we don't create objects that exceed the max value of the address space.  On aarch64 and riscv64 the align_down doesn't actually exclude anything since the config.addrspace_max became power-of-2 aligned.